cmd/tuf/server/main.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2022 The Sigstore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/sigstore/scaffolding/pkg/repo"
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/signals"
+	"sigs.k8s.io/release-utils/version"
+)
+
+var (
+	// Static data to include in the trust root.
+	rekorPubKey = flag.String("rekor-pubkey", "/var/run/tuf-secrets/rekor-pubkey", "Path to public key of Rekor server")
+	fulcioCert  = flag.String("fulcio-cert", "/var/run/tuf-secrets/fulcio-cert", "Path to the fulcio certificate")
+	ctPubKey    = flag.String("ct-pubkey", "/var/run/tuf-secrets/ct-pubkey", "Path to a CT Log public key")
+	// Name of the "secret" initial 1.root.json.
+	secretName = flag.String("rootsecret", "tuf-root", "Name of the secret to create for the initial root file")
+)
+
+func main() {
+	flag.Parse()
+
+	ns := os.Getenv("NAMESPACE")
+	if ns == "" {
+		panic("env variable NAMESPACE must be set")
+	}
+	ctx := signals.NewContext()
+
+	versionInfo := version.GetVersionInfo()
+	logging.FromContext(ctx).Infof("running create_repo Version: %s GitCommit: %s BuildDate: %s", versionInfo.GitVersion, versionInfo.GitCommit, versionInfo.BuildDate)
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		logging.FromContext(ctx).Panicf("Failed to get InClusterConfig: %v", err)
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		logging.FromContext(ctx).Panicf("Failed to get clientset: %v", err)
+	}
+
+	// Read the Rekor file
+	rekor, err := ioutil.ReadFile(*rekorPubKey)
+	if err != nil {
+		logging.FromContext(ctx).Panicf("Failed to read Rekor pubkey %s: %v", *rekorPubKey, err)
+	}
+
+	fulcio, err := ioutil.ReadFile(*fulcioCert)
+	if err != nil {
+		logging.FromContext(ctx).Panicf("Failed to read Fulcio cert %s: %v", *fulcioCert, err)
+	}
+
+	ct, err := ioutil.ReadFile(*ctPubKey)
+	if err != nil {
+		logging.FromContext(ctx).Panicf("Failed to read ctPubkey %s: %v", *ctPubKey, err)
+	}
+
+	// Create a new TUF root with the listed artifacts.
+	local, dir, err := repo.CreateRepo(ctx, fulcio, rekor, ct)
```

Comment:
Question: What would we expect whenever the server restarts ? It is okay to recreate the repository again and again.

Reply:
Good point, we can recreate the server, but this means you would need to refetch the rootsecret reinitialize Cosign since it would create a new initial root.

Reply:
Oh, it's not idempotent based on the same inputs?

I'm working on cleaning up much of the other jobs in #263 so if y'all think it's ok to checkpoint this and I can address these issues in that PR? I could tar the repository into a secret (or configmap if it's too big) and untar upon restart if it exists.